include/bmat8_impl.hpp

```diff
@@ -397,6 +397,13 @@ inline std::vector<uint8_t> BMat8::rows() const {
     return rows;
 }
 
+inline void BMat8::rows(std::array<uint8_t, 8>& rows) const {
+      rows.fill(0);
+      for (size_t i = 0; i < 8; ++i) {
+        rows[i] = static_cast<uint8_t>(_data << (8 * i) >> 56);
```

Comment:
There might be an even (slightly) faster way to do that using ` _mm_set_epi64x(_data, 0)` and the `_mm_extract_epi8` instruction. Actually some cast could do it, but we have to avoid some undefined behavior.

Reply:
This would be most excellent.